### PR TITLE
[TRAFFIC-9327] - Add `confluent network transit-gateway-attachment create` commands

### PR DESCRIPTION
**Pull request by syoung-cflt**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli. We will merge commits into main from this feature branch after all of the commands for network are implemented
* Add `confluent network transit-gateway-attachment create`

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests

Open Questions / Follow-ups
---------------------------

### CODE REVIEW

File: internal/network/command_transit_gateway_attachment_create.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	cobra.CheckErr(cmd.MarkFlagRequired("aws-transit-gateway"))
+	cobra.CheckErr(cmd.MarkFlagRequired("routes"))
+
+	return cmd
+}
+
+func (c *transitGatewayAttachmentCommand) create(cmd *cobra.Command, args []string) error {
+	name := args[0]
+
+	networkId, err := cmd.Flags().GetString("network")
```
Comment (brianstrauch) [inline internal/network/command_transit_gateway_attachment_create.go:45]:
```suggestion
	network, err := cmd.Flags().GetString("network")
```

Comment (syoung-cflt) [reply internal/network/command_transit_gateway_attachment_create.go]: Updated!

File: internal/network/command_transit_gateway_attachment_create.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		return err
+	}
+	networkId = network.GetId()
+
+	awsRamShareArn, err := cmd.Flags().GetString("aws-ram-share-arn")
+	if err != nil {
+		return err
+	}
+
+	awsTransitGatewayId, err := cmd.Flags().GetString("aws-transit-gateway")
```
[the remark this answers is not in the document]
Comment (syoung-cflt) [reply internal/network/command_transit_gateway_attachment_create.go]: Updated!

File: test/test-server/networking_handlers.go [hunk 3]
```diff
@@ -593,3 +595,40 @@ func handleNetworkingTransitGatewayAttachmentDelete(_ *testing.T, id string) htt
 		}
 	}
 }
+
+func handleNetworkingTransitGatewayAttachmentCreate(t *testing.T) http.HandlerFunc {
```
Comment (ejagade) [inline test/test-server/networking_handlers.go:599]: can we add a test case for error response from networking service

**Review by ejagade — APPROVED** [review]

File: test/test-server/networking_handlers.go (hunk 3 above)
Comment (syoung-cflt) [reply test/test-server/networking_handlers.go:599]: Added a few cases for error response

**Review by ejagade — APPROVED** [review]

**Pull request merged**